src/goal_gui.cpp

```diff
@@ -83,7 +83,7 @@ struct GoalListWindow : public Window {
 		}
 
 		y -= 2; // "Company specific goals:" line.
-		if (y < 0) return;
+		if (y < 0 || this->owner == INVALID_OWNER) return;
 
 		for (const Goal *s : Goal::Iterate()) {
 			if (s->company == this->window_number) {
```

Comment:
I think it's better to replicate the test used to populate the sublists https://github.com/OpenTTD/OpenTTD/blob/63b56bb76b4090fa3af611cdf74f668a63dc28ed/src/goal_gui.cpp#L209
```suggestion
		if (y < 0) return;

		for (const Goal *s : Goal::Iterate()) {
			if (s->company == this->window_number && s->company != INVALID_COMPANY) {
```

Reply:
It's not `s->company` that needs to test agains `INVALID_OWNER` / `INVALID_COMPANY`, it's either `this->owner` or `this->window_number`

Reply:
Nevermind, you're right.